aten/src/ATen/native/TensorShape.cpp

```diff
@@ -34,6 +34,15 @@ Tensor _shape_as_tensor(const Tensor& self) {
   return at::tensor(self.sizes(), options);
 }
 
+Tensor& set_(Tensor& result) {
+  Storage storage(result.dtype(), 0, c10::GetAllocator(result.device().type()), true);
```

Comment:
Isn't the storage supposed to be a nullptr from the previous signature in case it is not given by the user?

Reply:
yes, but we can't don't actually support passing nullptrs in this way through the ATen->TH binding layer.

So what this PR attempts to do instead is to create a Storage in the same way the TH code does if you pass in a nullptr (you can't have nullptr Storage, so TH does this logic already).

Reply:
Right, the fix looks like the right thing to do :)